File: OOP-Java/Java Basics/classes_and_objects_basics/Student.java
package classes_and_objects_basics;

public class Student {
    private int rollNumber;     //All the property is initialised with 0
    String name;        //String store address so it store null

    public Student(String n, int rollno) {
        name = n;
        rollNumber = rollno;
    }

//    public Student(){     //can't do this with custom constructor
//        rollNumber = 10;
//    }

    public Student(String n) {
        name = n;
    }

    public void setRollNumber(int rn) {
        if (rn <= 0) {
            return;
        }
        rollNumber = rn;
    }

    public int getRollNumber() {
        return rollNumber;
    }

    public void print() {
        System.out.println(name + " " + rollNumber);
    }
}

/*
   (Access modifier)
   Default- Within package
   Public - Anywhere
   Private - Only within class
 */

 /*
    Specifier	class	subclass	package     world
    private      yes	   no	      no	     no
    protected	 yes	   yes	      yes        no
    public	     yes	   yes	      yes	     yes
    (none)       yes       no	      yes	     no
  */

File: OOP-C++/Basics/Part1/Basics/3 Default Constructor/Student.cpp
class Student {

public:
	int rollNumber;

private:
	int age;

public:
	// Default constructor
	Student() {
		cout << "Constructor called ! "<< endl;
	}

	void display() {
		cout << "Age of student: " << age << endl <<"Roll no: " << rollNumber << endl;
		// cout << this.age << " " << this.rollNumber << endl;
	}

	int getAge() {
		return age;
	}

	void setAge(int a, int password) {
		if(password != 123) {
			return;
		}
		if(a < 0) {
			return;
		}
		age = a;
	}
};

/*
   By default - access specifier is private
 */
